realm/realm-library/src/androidTest/java/io/realm/RealmSchemaTests.java

```diff
@@ -322,4 +324,43 @@ public void remove_shouldRemoveInfoFromPKTable() {
         objectSchema.addPrimaryKey(PrimaryKeyAsString.FIELD_PRIMARY_KEY);
         assertEquals(PrimaryKeyAsString.FIELD_PRIMARY_KEY, objectSchema.getPrimaryKey());
     }
+
+    @Test
+    public void remove_then_cancel() {
+        if (type == SchemaType.IMMUTABLE) {
+            return;
+        }
+
+        realmSchema.create("foo");
+        // getSchemaForClass is an internal method, but used from DynamicRealmObject and RealmQuery
+        realmSchema.getSchemaForClass("foo"); // make cache entry
```

Comment:
No. `realmSchema.get()` is not using the cache. `getSchemaForClass(Class)` also has the same issue, but that method is only used for typed models and no classes are removed actually.